public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx

```diff
@@ -203,83 +208,84 @@ export class DetectionVisualEditor extends React.Component<
     return dump(compiledDetection);
   };
 
-  private updateDatumInState = (
-    selectionIdx: number,
-    dataIdx: number,
-    newDatum: Partial<SelectionData>
-  ) => {
+  private validateDatum = (selections: Selection[]) => {
```

Comment:
since we are validating multiple data entries we can call this `validateData`

Reply:
It makes sense, and will be refactored.